client/src/components/notifications/UserSharedDocumentNotification.vue

```diff
@@ -0,0 +1,122 @@
+
+<template>
+  <notification-item
+    :notification="notification"
+    id="user-shared-document-notification"
+  >
+    <div class="notification-content">
+      <user-avatar-name
+        class="notification-avatar"
+        :user-avatar="userInfo ? userInfo.humanHandle.label : ''"
+      />
+      <div class="notification-details">
+        <ion-text class="notification-details__message body">
+          <i18n-t
+            keypath="notification.userSharedDocument"
+            scope="global"
+          >
+            <template #name>
+              <strong>{{ userInfo ? userInfo.humanHandle.label : '' }}</strong>
+            </template>
+          </i18n-t>
+        </ion-text>
+        <ion-text class="notification-details__time body-sm">
+          <span>{{ formatTimeSince(notification.time, '', 'short') }}</span>
+        </ion-text>
+      </div>
+    </div>
+    <div
+      class="notification-file"
+      @click="goToEnclosingFolder"
+    >
+      <ms-image
+        :image="getFileIcon(notificationData.fileName)"
+        class="default-state file-icon"
+      />
+      <ms-image
+        :image="Folder"
+        class="hover-state file-icon"
+      />
+      <ion-text class="notification-file__name subtitles-sm">
+        <span class="default-state">{{ notificationData.fileName }}</span>
+        <span class="hover-state">{{ $t('notificationCenter.openEnclosingFolder') }}</span>
+      </ion-text>
+      <ion-text class="notification-file__size default-state subtitles-sm">
+        {{ formatFileSize(notificationData.fileSize) }}
+      </ion-text>
+    </div>
+  </notification-item>
+</template>
+
+<script setup lang="ts">
+import { formatTimeSince } from '@/common/date';
+import { formatFileSize, getFileIcon } from '@/common/file';
+import { Folder, MsImage } from '@/components/core/ms-image';
+import NotificationItem from '@/components/notifications/NotificationItem.vue';
+import UserAvatarName from '@/components/users/UserAvatarName.vue';
+import { UserInfo, getUserInfo, getWorkspaceName } from '@/parsec';
+import { UserSharedDocumentData } from '@/services/informationManager';
+import { Notification } from '@/services/notificationManager';
+import { IonText, popoverController } from '@ionic/vue';
+import { Ref, onMounted, ref } from 'vue';
+
+const userInfo: Ref<UserInfo | null> = ref(null);
+const workspaceName = ref('');
+const props = defineProps<{
+  notification: Notification;
+}>();
+
+onMounted(async () => {
+  const resultWorkspace = await getWorkspaceName(notificationData.workspaceId);
+  const resultUser = await getUserInfo(notificationData.userId);
+
+  if (resultWorkspace.ok) {
+    workspaceName.value = resultWorkspace.value;
+  }
+
+  if (resultUser.ok) {
+    userInfo.value = resultUser.value;
+  }
+});
+
+async function goToEnclosingFolder(): Promise<void> {
+  await popoverController.dismiss();
+  console.log('Go to enclosing folder');
```

Comment:
You can use `await navigateToWorkspace(notificationData.workspaceId, notificationData.filePath)` to go to the folder (it won't work correctly with the mocks since the folders are generated randomly).